lib/withOnyx.js

```diff
@@ -110,9 +123,8 @@ export default function (mapOnyxToState, shouldDelayUpdates = false) {
             componentWillUnmount() {
                 // Disconnect everything from Onyx
                 _.each(mapOnyxToState, (mapping) => {
-                    const key = Str.result(mapping.key, this.props);
-                    const connectionID = this.activeConnectionIDs[key];
-                    Onyx.disconnect(connectionID, key);
+                    const key = Str.result(mapping.key, {...this.props, ...getOnyxDataFromState(this.state, mapOnyxToState)});
```

Comment:
This is one change that is different than the original PR. I realized that the `unmount()` was still trying to use the old method for getting the key name. This would probably lead to somewhat of a memory leak because it could cause Onyx to keep a bunch of connections that are never disconnected from when components unmount.

Reply:
I can give it a try. That's a good hunch

Reply:
No, it doesn't appear to. I think the changes for batched updates was probably the biggest help for preventing the performance problem we saw last time.

Reply:
>  I just feel like this might cause more issues down the road.

Are you referring to the change in `onUnmount()` specifically, or this PR in general?

Reply:
To dig into this further, I tried the following things:

1. Reverted Onyx to the commit where the original PR was reverted. Then I copied these `withOnyx` changes to the code and tried it out. **Result:** I was not able to duplicate the original performance problem.
2. Reverted Onyx to the commit of the original PR being merged, and tried it out in the app. **Result:** I experienced the following fatal error (caused by the `initialValue` change). After fixing _only that fatal error_ I was still not able to reproduce the original performance problem, though I did see that the report switching was taking about 2-3x longer.

![image](https://github.com/Expensify/react-native-onyx/assets/1228807/d6902217-be2c-4e14-b555-f0314c5f328f)
![image](https://github.com/Expensify/react-native-onyx/assets/1228807/b93549a1-48e8-43f3-9a83-3d0a64ef4b2b)

I'm not sure if this fatal error was what we experienced when the original PR was deployed. We definitely witnessed the app hanging and no one could report any JS console errors or record performance profiles, but that was about it.

**Conclusion:** I am feeling pretty good about the changes in this PR in general. The worst that can happen is we have to do another quick revert, which would stink, but it's also not the end of the world.